src/lib.rs

```diff
@@ -717,6 +717,325 @@ where
     }
 }
 
+macro_rules! impl_primitive_ops_ratio {
+    (Ratio<T> for $primitive:ident) => {
+        impl<T> Add<Ratio<T>> for $primitive
+        where
+            T: Clone + Integer + ::core::convert::From<$primitive>,
```

Comment:
I think adding a blanket `impl<T>` may be a breaking change, e.g. if someone already had their own `impl Add<Ratio<Foo>> for i32`.  Honestly, I still get confused about exactly what the coherence rules would allow, but I think this would be legal for someone else to do, and your change would break it.

But regardless, I don't think we want implicit conversion between primitives.  Note that the op implementations in the standard library don't do this conversion either.  Idiomatic Rust is that primitive types should be explicitly converted as needed.

On the other hand, `BigInt` and `BigUint` *do* implement ops for the primitive types, and this is for performance reasons.  They can entirely avoid allocating a new bigint buffer in most cases.  However, here you defeat that idea if you use `T::from` here, because that forces a new allocation for that bigint `T`.

So I think the implementations we do want are:

- `impl Add<Ratio<$primitive>> for $primitive`
- `impl Add<BigRational> for $primitive`
- `impl Add<BigRational> for BigInt`
- could do `Ratio<BigUint>` too, but we're probably lacking in a lot of places for that...
- `&` variations of all of these -- ref+value, value+ref, ref+ref